src/ProducerEventProcessor/ProducerEventProcessor.cpp

```diff
@@ -536,9 +593,14 @@ void ProducerEventProcessorImpl::ProcessThreadNamesSnapshotAndTransferOwnership(
 
 void ProducerEventProcessorImpl::ProcessThreadStateSliceAndTransferOwnership(
     ThreadStateSlice* thread_state_slice) {
-  ClientCaptureEvent event;
-  event.set_allocated_thread_state_slice(thread_state_slice);
-  client_capture_event_collector_->AddEvent(std::move(event));
+  if (thread_state_slice->switch_out_or_wakeup_callstack_status() ==
+      ThreadStateSlice::kNoCallstack) {
+    ClientCaptureEvent event;
+    event.set_allocated_thread_state_slice(thread_state_slice);
+    client_capture_event_collector_->AddEvent(std::move(event));
+    return;
+  }
+  MergeThreadStateSliceWithCallstack(thread_state_slice);
```

Comment:
This should also be called `...AndTransferOwnership` to signal that it does `set_allocated_thread_state_slice`, otherwise this looks like a leak of the `ThreadStateSlice`.

Reply:
Done.